sky/serve/load_balancer.py

```diff
@@ -77,23 +78,101 @@ def _sync_with_controller(self):
+                        # the async generator to avoid the client being closed
+                        # before the response is consumed. However, we still
+                        # need the response status code and headers to construct
+                        # the StreamingResponse, which is only available after
+                        # the client is constructed. We yield them first here.
+                        # TODO(tian): Investigate a way to not directly yielding
+                        # the response status code and headers.
+                        yield response.status_code
+                        yield dict(response.headers)
+                        try:
+                            async for chunk in response.aiter_bytes():
+                                yield chunk
+                        except Exception as e:  # pylint: disable=broad-except
+                            yield f'Error: {str(e)}'
+                        finally:
+                            await response.aclose()
+
+            content = stream_response()
+            status_code = await content.__anext__()
+            headers = await content.__anext__()
+            return fastapi.responses.StreamingResponse(content=content,
+                                                       status_code=status_code,
+                                                       headers=headers)
+        except (httpx.RequestError, httpx.HTTPStatusError) as e:
+            logger.error(f'Error when proxy request to {path}: '
+                         f'{common_utils.format_exception(e)}')
+            return None
+
+    async def _proxy_with_retries(
+            self, request: fastapi.Request) -> fastapi.responses.Response:
+        """Try to proxy the request to the endpoint replica with retries."""
+        self._request_aggregator.add(request)
+        # TODO(tian): Finetune backoff parameters.
+        backoff = common_utils.Backoff(initial_backoff=1)
+        # SkyServe supports serving on Spot Instances. To avoid preemptions
+        # during request handling, we add a retry here.
+        retry_cnt = 0
+        while True:
+            retry_cnt += 1
+            ready_replica_url = self._load_balancing_policy.select_replica(
+                request)
+            if ready_replica_url is None:
+                raise fastapi.HTTPException(
+                    status_code=503,
+                    detail='No ready replicas. '
+                    'Use "sky serve status [SERVICE_NAME]" '
+                    'to check the replica status.')
+            response = await self._proxy_request_to(ready_replica_url, request)
+            if response is not None:
+                return response
+            # TODO(tian): Fail fast for errors like 404 not found.
+            if retry_cnt == constants.LB_MAX_RETRY:
+                raise fastapi.HTTPException(
+                    status_code=500,
```

Comment:
Good to comment for what does 500 mean in HTTP response.

Reply:
Also, can we include the exception raised by the underlying server in the detail?